lldb/source/Plugins/Language/Swift/SwiftLanguage.cpp

```diff
@@ -1793,6 +1793,17 @@ SwiftLanguage::GetDemangledFunctionNameWithoutArguments(Mangled mangled) const {
   return mangled_name;
 }
 
+bool SwiftLanguage::IsInterestingCtxForLineBreakpoint(
+    const SymbolContext &sc) const {
+  // If we don't have a function, conservatively return true.
+  if (sc.function)
+    return true;
+  // In async functions, ignore "await resume" funclets, these only deallocate
+  // the async context and task_switch back to user code.
+  auto demangled = sc.function->GetMangled().GetDemangledName(&sc);
+  return demangled.GetStringRef().contains("await resume");
```

Comment:
A much better way to implement this would be to walk the demangle tree.
Also this function will incorrectly return true if I had a function that takes an "await resume" function pointer as argument. By just checking the outermost nodes of the demangle tree you can address both problems.

Look at the output of `swift-demangle -expand <mangled name>` and copy any of the functions in TypeSystemSwiftTypeRef.cpp.

Reply:
> will incorrectly return true if I had a function that takes an "await resume" function pointer as argument

Is this possible? these functions are created by the corosplitter

Reply:
Probably not, but who knows what future runtime optimizations will be trying to do.

Reply:
In SwiftLanguageRuntimeNames.cpp there is IsSwiftAsyncFunctionSymbol which you might be able to adapt.